### PR TITLE
whatsnew - branded section icons

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR adds icon branding to the `whatsnew` section headers.

Some of the icons align with those already chosen for the associated [issues icon templates](https://github.com/SciTools/iris/issues/new/choose).

If and when #3850 is merged, I'll align the `Announcements` section appropriately.

To help reviewers, the rendered `readthedocs` changes of this PR are available:
- [latest whatsnew](https://my-iris.readthedocs.io/en/latest/whatsnew/latest.html).
- [Contribution categories](https://my-iris.readthedocs.io/en/latest/developers_guide/documenting/whats_new_contributions.html#contribution-categories)
---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/pulls.html#the-iris-check-list)

### CODE REVIEW

File: docs/iris/src/whatsnew/latest.rst.template [hunk 1]
```diff
@@ -10,37 +10,50 @@ This document explains the changes made to Iris for this release
...
-Features
-========
+🎉 Announcements
+================
 
 * N/A
 
 
-Bugs Fixed
+⚙ Features
```
Comment (trexfeathers) [inline docs/iris/src/whatsnew/latest.rst.template:19]:
```suggestion
✨ Features
```
The current cog/gear is rendering as a compass symbol in some places. (I appreciate any change to this one might also require a change to [ISSUE_TEMPLATE/feature-request.md](https://github.com/SciTools/iris/blob/6b675e34d9aef7b29d84ce9911d76f46a2d46230/.github/ISSUE_TEMPLATE/feature-request.md))

File: docs/iris/src/whatsnew/latest.rst.template [hunk 2]
```diff
@@ -10,37 +10,50 @@ This document explains the changes made to Iris for this release
    :depth: 3
 
 
-Features
-========
+🎉 Announcements
```
Comment (trexfeathers) [inline docs/iris/src/whatsnew/latest.rst.template:13]:
```suggestion
📢 Announcements
```

File: docs/iris/src/whatsnew/latest.rst [hunk 3]
```diff
@@ -10,8 +10,14 @@ This document explains the changes made to Iris for this release
    :depth: 3
 
 
-Features
-========
+🎉 Announcements
```
Comment (trexfeathers) [inline docs/iris/src/whatsnew/latest.rst:13]:
```suggestion
📢 Announcements
```

File: docs/iris/src/whatsnew/latest.rst [hunk 4]
```diff
@@ -131,8 +137,8 @@ Incompatible Changes
   duplicate coordinate errors in certain circumstances. (:pull:`3718`)
 
 
-Deprecations
-============
+🚫 Deprecations
```
Comment (trexfeathers) [inline docs/iris/src/whatsnew/latest.rst:140]:
```suggestion
🗑️ Deprecations
```
or
```suggestion
🔥 Deprecations
```
... something that's more distinct from the ⛔ of `Incompatible Changes`

File: docs/iris/src/whatsnew/latest.rst.template [hunk 5]
```diff
@@ -10,37 +10,50 @@ This document explains the changes made to Iris for this release
...
+⛔ Incompatible Changes
+=======================
+
 
 * N/A
 
 
-Documentation
-=============
+🚫 Deprecations
```
Comment (trexfeathers) [inline docs/iris/src/whatsnew/latest.rst.template:38]:
```suggestion
🗑️ Deprecations
```
or
```suggestion
🔥 Deprecations
```
... something that's more distinct from the ⛔ of `Incompatible Changes`

File: docs/iris/src/whatsnew/latest.rst [hunk 6]
```diff
@@ -10,8 +10,14 @@ This document explains the changes made to Iris for this release
...
 
-Features
-========
+🎉 Announcements
+================
+
+* N/A
+
+
+⚙ Features
```
Comment (trexfeathers) [inline docs/iris/src/whatsnew/latest.rst:19]:
```suggestion
✨ Features
```
The current cog/gear is rendering as a compass symbol in some places. (I appreciate any change to this one might also require a change to [ISSUE_TEMPLATE/feature-request.md](https://github.com/SciTools/iris/blob/6b675e34d9aef7b29d84ce9911d76f46a2d46230/.github/ISSUE_TEMPLATE/feature-request.md))

**Review by trexfeathers — CHANGES_REQUESTED** [review]

File: docs/iris/src/whatsnew/latest.rst.template (hunk 1 above)
Comment (bjlittle) [reply docs/iris/src/whatsnew/latest.rst.template:19]: It is actually **gears**... apparently.

However, how can I neglect to substitute it with sparkles. Nice one, let's go with that :+1:

File: docs/iris/src/whatsnew/latest.rst.template (hunk 2 above)
Comment (bjlittle) [reply docs/iris/src/whatsnew/latest.rst.template:13]: Yup, love it. I'm sold on that too :smile:

File: docs/iris/src/whatsnew/latest.rst (hunk 4 above)
Comment (bjlittle) [reply docs/iris/src/whatsnew/latest.rst:140]: Given the choice, I'd go with some flamage. Great suggestion :+1:

Comment (bjlittle) [reply docs/iris/src/whatsnew/latest.rst:140]: @trexfeathers ... now you've started something :wink: 

Going on the flame theme, I've changed `Incompatible Changes` to 💣 (bomb) ... almost suggested 💥 (collision), but again that's too similar to 🔥 `Deprecations`

**Comment by trexfeathers** [comment]
Nice, we seem to have converted to the 'fun logo pack', which I am fully in favour of!

**Review by trexfeathers — APPROVED** [review]

**Comment by bjlittle** [comment]
@trexfeathers Superb! Thanks :+1:

**Pull request merged**